### PR TITLE
Initialize ViewComponent::Config with defaults before framework load

### PR DESCRIPTION
**Pull request by boardfish**
This PR introduces `ViewComponent::Config.current`, which should not be used directly by consumers of ViewComponent, but is used as the common object from which `ViewComponent::Base` and `Rails.application.config.view_component` will load their config once each is ready. It'll make config available from the very start with the intent of not relying upon the framework to find these options - we can manually set anything reliant on Action View, for example, within the app or engine config.

#1659 made the config object empty before ViewComponent and Rails were initialised. This would mean that code within ViewComponent that relied on configured defaults would break, resulting in issues like #1741. Since ViewComponent::Config can be loaded independently of Action View, this should hopefully no longer be the case.

If you were affected by #1741, please test against this branch to see if it fixes your issue. Thanks for your patience!

### CODE REVIEW

**Review by camertron — COMMENTED** [review]
Hey @boardfish thanks for working on this. I don't understand how `current` get set - there doesn't appear to be any call to it?

**Comment by Spone** [comment]
> Hey @boardfish thanks for working on this. I don't understand how `current` get set - there doesn't appear to be any call to it?

I think the `defaults` method is called here:

```rb
class_attribute :current, default: defaults, instance_predicate: false
```

**Comment by boardfish** [comment]
Hey folks, is there anything else you'd like me to do with this so we can get it shipped? I'm aware there are now multiple issues that this will fix.

File: lib/view_component/preview.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 module ViewComponent # :nodoc:
   class Preview
-    include Rails.application.routes.url_helpers if defined?(Rails.application.routes.url_helpers)
+    include Rails.application.routes.url_helpers if Rails && Rails.application.present?
```
Comment (BlakeWilliams) [inline lib/view_component/preview.rb:7]: Do we still need the `defined?` check to support non-Rails environments?

**Review by BlakeWilliams — APPROVED** [review]

File: lib/view_component/preview.rb (hunk 1 above)
Comment (boardfish) [reply lib/view_component/preview.rb:7]: Yeah, I think this must've fallen through in a rebate 🧐 I'll make sure to fix it before it gets merged.

**Comment by camertron** [comment]
@Spone looks like there's a coverage issue as well 😦

**Comment by Spone** [comment]
Do you know what's happening with the coverage @boardfish?

**Comment by boardfish** [comment]
I think it wants `ViewComponent::Base#config=` to be tested, but I think that method might even be able to be removed.

**Comment by ryansdwilson** [comment]
Thank you all for your work on this! I ran into this issue today and went down a rabbit hole trying to figure it out. I really appreciate all the work you're doing to maintain VC.

**Review by camertron — APPROVED** [review]

**Comment by boardfish** [comment]
Thanks for wrapping this up @camertron, and sorry I haven't been able to dedicate the time to it! 😅 🙏

**Pull request merged**